nix/lib/purescript.nix

```diff
@@ -1,8 +1,8 @@
 { stdenv
 , lib
 , cacert
-, nodejs-10_x
-, nodePackages_10_x
+, nodejs
+, nodePackages
```

Comment:
I wonder if we should perhaps have an overlay pinning nodejs/nodePackages to a version such that we would have it in one central place like..

```
self: super:
{
  node = {
    nodejs =  super.nodejs-14_x;
    nodejsPackages = super.nodePackages;
  };
}
```

Of course we can also just leave it as `nodejs` and `nodePackages` (i just realized nixpkgs doesn't have different nodePackages versions anymore anyway) which i assume is just always going to point to the latest version and if that does break somehow when it changes we just have to fix it.

Reply:
it is specifically for these purescript projects though, it may be in the future that we build other node stuff with other node versions. It _should_ be using the nixpkgs default but this says that it specifically doesn't for purescript.

Reply:
My though was that *this* builder should just assume `nodejs` and `nodePackages`, and we can override them in the `callPackage` call if we need to pass a particular version.

Reply:
oh yes, i should have been more explicit here: i totally agree that purescript.nix should *receive* `nodejs` and `nodePackages` without making any version decisions on its own.

I was just thinking that since `nodejs` is like a moving target that might change to different versions i was wondering if we should somehow pinpoint which specific nodejs version we want to use everywhere.

Then again maybe i'm over-engineering this. I did do that in a different project for both nodejs and python because both were used a lot over the project and it was easy to accidentally use different versions which we wanted to avoid.